src/Storefront/Theme/ThemeService.php

```diff
@@ -632,4 +690,17 @@ private function isAsyncCompilation(Context $context): bool
 
         return $this->configService->get(self::CONFIG_THEME_COMPILE_ASYNC) && !$context->hasState(self::STATE_NO_QUEUE);
     }
+
+    private function buildSnippetKey(string $themeTechnicalName, bool $isHelpText, string ...$parts): string
+    {
+        return implode(
+            '.',
+            [
+                'sw-theme',
+                XmlParserUtils::kebabCaseToCamelCase($themeTechnicalName),
```

Comment:
The XmlParserUtils is only for the App system usage.

Symfony has already functionalities for manipulating strings: https://symfony.com/doc/current/string.html

Reply:
Exchanged it with Symfonys new "u" function